WcaOnRails/app/models/role.rb

```diff
@@ -2,4 +2,28 @@
 
 class Role < ApplicationRecord
   belongs_to :user
+  belongs_to :group, class_name: "UserGroup"
+  belongs_to :metadata, polymorphic: true, optional: true
+
+  delegate :group_type, to: :group
+
+  enum :team_status, {
```

Comment:
Does the `roles` table have a `team_status` column?
The `enum` construct is specifically made by **Rails** (not Ruby in general!) to enumerate string (or integer) values in a DB column. It feels like a waste to use this for regular class constants if there is no column restriction to check.

Reply:
In case of doubt, just use `TEAM_STATUS_LEADER = "leader"` and similar as top-level class constants.

Reply:
Actually the team_status should belong to the `team_role_metadata` which is yet to come. For now, I've created top-level class constant in `user.rb` since this is currently used in that class.